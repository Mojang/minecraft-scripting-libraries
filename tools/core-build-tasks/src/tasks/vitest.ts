import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
