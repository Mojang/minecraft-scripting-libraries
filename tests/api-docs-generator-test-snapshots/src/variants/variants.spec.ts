// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Variants', () => {
    it('Properly generates documentation for modules which contain variants', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
