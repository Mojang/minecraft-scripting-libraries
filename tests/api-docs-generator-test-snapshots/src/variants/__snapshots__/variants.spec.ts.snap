// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Variants > Properly generates documentation for modules which contain variants > # Of Files Generated 1`] = `4`;

exports[`Variants > Properly generates documentation for modules which contain variants > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: variants-module
    items:
    - name: variants-module
      href: variants-module/variants-module.md
    - name: ExampleClass
      href: variants-module/ExampleClass.md
"
`;

exports[`Variants > Properly generates documentation for modules which contain variants > msdocs/scriptapi/variants-module/ExampleClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: variants-module.ExampleClass Class
description: Contents of the variants-module.ExampleClass class.
---
# ExampleClass Class

## Properties

### **propertyFunctionThatReturnsVariant**
\`propertyFunctionThatReturnsVariant: (arg0: number) => (number | string);\`

Type: (arg0: *number*) => (*number* | *string*)

Notes:
  - This property can't be used in read-only mode.

### **propertyFunctionThatReturnsVariantWithClosureThatReturnsVariant**
\`propertyFunctionThatReturnsVariantWithClosureThatReturnsVariant: (arg0: number) => ((arg0: number) => (number | string) | string);\`

Type: (arg0: *number*) => ((arg0: *number*) => (*number* | *string*) | *string*)

Notes:
  - This property can't be used in read-only mode.

## Methods
- [exampleMethod](#examplemethod)

### **exampleMethod**
\`
exampleMethod(arg1: (arg0: number) => ((arg0: number) => (number | string) | string), arg2: number): void
\`

#### **Parameters**
- **arg1**: (arg0: *number*) => ((arg0: *number*) => (*number* | *string*) | *string*)
- **arg2**: *number*
  
Notes:
- This function can't be called in read-only mode.
"
`;

exports[`Variants > Properly generates documentation for modules which contain variants > msdocs/scriptapi/variants-module/variants-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: variants-module Module
description: Contents of the variants-module module
---
# \`variants-module\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "variants-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [ExampleClass](ExampleClass.md)
"
`;

exports[`Variants > Properly generates documentation for modules which contain variants > typescript/variants-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "variants-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export class ExampleClass {
    private constructor();
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    propertyFunctionThatReturnsVariant: (arg0: number) => number | string;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    propertyFunctionThatReturnsVariantWithClosureThatReturnsVariant: (
        arg0: number
    ) => (arg0: number) => (number | string) | string;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    exampleMethod(arg1: (arg0: number) => (arg0: number) => (number | string) | string, arg2: number): void;
}
"
`;
